src/Parlot/Cursor.cs

```diff
@@ -123,13 +123,15 @@ public void AdvanceNoNewLines(int offset)
             if (newOffset > _textLength - 1)
```

Comment:
```suggestion
            var length = _textLength - 1;
            if (newOffset > length)
```

Reply:
I have updated the changes as suggested.